handlers.go

```diff
@@ -229,6 +241,31 @@ func (m *MockOIDC) validateCodeGrant(rw http.ResponseWriter, req *http.Request)
 	return session, true
 }
 
+func (m *MockOIDC) validateCodeChallenge(rw http.ResponseWriter, req *http.Request, session *Session) bool {
+	if session.CodeChallenge == "" || session.CodeChallengeMethod == "" {
+		return true
+	}
+
+	codeVerifier := req.Form.Get("code_verifier")
+	if codeVerifier == "" {
+		errorResponse(rw, InvalidGrant, "Invalid code verifier. Expected code but client sent none.", http.StatusUnauthorized)
```

Comment:
Note that invalid grant is the correct error code here:
> If the values are not
 equal, an error response indicating "invalid_grant" as described in
 [Section 5.2 of [RFC6749]](https://datatracker.ietf.org/doc/html/rfc6749#section-5.2) MUST be returned.